homeassistant/components/mqtt/client.py

```diff
@@ -877,6 +877,25 @@ async def _async_perform_unsubscribes(self) -> None:
 
         await self._wait_for_mid(mid)
 
+    async def _resubscribe_and_publish_birth_message(
+        self, birth_message: PublishMessage
+    ) -> None:
+        """Resubscribe to all topics and publish birth message."""
+        await self._async_perform_subscriptions()
+        await self._ha_started.wait()  # Wait for Home Assistant to start
+        await self._discovery_cooldown()  # Wait for MQTT discovery to cool down
+        # Check subscriptions again to ensure we are subscribed
+        # in case anything pending was added while HA was starting
+        await self._async_perform_subscriptions()
```

Comment:
I am not sure about this. If we subscribe the wildcard `homeassistant/#` then that will ensure we get all messages from that moment. As soon as we trigger the birth message we will trigger other applications to update

Reply:
Lets be a bit more conservative here since the release is tomorrow

removed in 8aae84f3920f1dfaf5d2054b246dc3fc111281e6